crates/net/src/http/mod.rs

```diff
@@ -15,434 +15,10 @@
-    /// [HTTP status code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status) from
-    /// `Response::status`.
-    ///
-    /// For example, this would be 'OK' for a status code 200, 'Continue' for 100, or 'Not Found'
-    /// for 404.
-    pub fn status_text(&self) -> String {
-        self.response.status_text()
-    }
-
-    /// Gets the headers.
-    pub fn headers(&self) -> Headers {
-        Headers::from_raw(self.response.headers())
-    }
-
-    /// Has the response body been consumed?
-    ///
-    /// If true, then any future attempts to consume the body will error.
-    pub fn body_used(&self) -> bool {
-        self.response.body_used()
-    }
-
-    /// Gets the body.
-    pub fn body(&self) -> Option<ReadableStream> {
-        self.response.body()
-    }
-
-    /// Gets the raw [`Response`][web_sys::Response] object.
-    pub fn as_raw(&self) -> &web_sys::Response {
-        &self.response
-    }
-
-    /// Reads the response to completion, returning it as `FormData`.
-    pub async fn form_data(&self) -> Result<FormData, Error> {
-        let promise = self.response.form_data().map_err(js_to_error)?;
-        let val = JsFuture::from(promise).await.map_err(js_to_error)?;
-        Ok(FormData::from(val))
-    }
-
-    /// Reads the response to completion, parsing it as JSON.
-    #[cfg(feature = "json")]
-    #[cfg_attr(docsrs, doc(cfg(feature = "json")))]
-    pub async fn json<T: DeserializeOwned>(&self) -> Result<T, Error> {
-        serde_json::from_str::<T>(&self.text().await?).map_err(Error::from)
-    }
-
-    /// Reads the response as a String.
-    pub async fn text(&self) -> Result<String, Error> {
-        let promise = self.response.text().unwrap();
-        let val = JsFuture::from(promise).await.map_err(js_to_error)?;
-        let string = js_sys::JsString::from(val);
-        Ok(String::from(&string))
-    }
-
-    /// Gets the binary response
-    ///
-    /// This works by obtaining the response as an `ArrayBuffer`, creating a `Uint8Array` from it
-    /// and then converting it to `Vec<u8>`
-    pub async fn binary(&self) -> Result<Vec<u8>, Error> {
-        let promise = self.response.array_buffer().map_err(js_to_error)?;
-        let array_buffer: ArrayBuffer = JsFuture::from(promise)
-            .await
-            .map_err(js_to_error)?
-            .unchecked_into();
-        let typed_buff: Uint8Array = Uint8Array::new(&array_buffer);
-        let mut body = vec![0; typed_buff.length() as usize];
-        typed_buff.copy_to(&mut body);
-        Ok(body)
-    }
-}
-
-impl fmt::Debug for Response {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        f.debug_struct("Response")
-            .field("url", &self.url())
-            .field("redirected", &self.redirected())
-            .field("status", &self.status())
-            .field("headers", &self.headers())
-            .field("body_used", &self.body_used())
-            .finish()
-    }
-}
+pub use request::Request;
```

Comment:
```suggestion
#[doc(inline)]
pub use request::Request;
```

We should copy the documentation from the http crate to gloo-net.